account_invoice_change_currency/models/account_change_currency.py

```diff
@@ -0,0 +1,174 @@
+# Copyright 2017-2018 Vauxoo
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import api, fields, models
+from odoo.tools import float_compare
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+    original_price_unit = fields.Monetary(
+        default=lambda self: self.price_unit,
+        help="""
+        Save price unit from every line in the invoice when the currency back to
+        original state""",
+    )
+
+    def create(self, values):
+        lines = super().create(values)
+        for line in lines:
+            line.write({"original_price_unit": line.price_unit})
```

Comment:
Don't trigger a write just after a create, you only need to update values:

```suggestion
        values.setdefault("original_price_unit", values.get("price_unit"))
```

Reply:
In this case I could not apply this change because of inheritance issues.